### ISSUE TITLE
security_monkey should track lambda resource policies

### ISSUE THREAD

**Issue by monkeysecurity** [Netflix/security_monkey#351]
Lambda resource policies aren't displayed in the console but are very important.

Track lambda functions and their resource policies.
Alert on cross account access.

Currently, it seems that cross account principals in a lambda resource policy can only be granted to the entire account, and not to a specific IAM role within that account.  Until that is cleared up, it's probably a good idea to avoid cross account lambda invocation all together.

Wide open cross account lambda resource policy:

```
{
  "Statement": [
    {
      "Sid": "1",
      "Resource": "arn:aws:lambda:<region>:<accountnumber>:function:<name>",
      "Principal": {
        "AWS": "arn:aws:iam::<other_account_number>:root"
      },
      "Effect": "Allow",
      "Action": "lambda:InvokeFunction"
    }
  ],
  "Version": "2012-10-17",
  "Id": "default"
}
```

**Comment by KevinHock** [comment]
What steps would you take if you were going to add this to security monkey? Maybe I can

**Comment by monkeysecurity** [comment]
1) PR to the cloudaux library adding lambda:
https://github.com/Netflix-Skunkworks/cloudaux

2) PR to security_monkey to add a watcher that uses cloudaux.  (See IAM Role/IAM User/S3 for examples)

3) [Optional] PR to security_monkey to add an auditor for lambda.

**Comment by monkeysecurity** [comment]
Fixed in SM #771 
Fixed in https://github.com/Netflix-Skunkworks/cloudaux/pull/41

Turns out Lambda resource policies are pretty complex.  The PR tracks alias, version, and default policies.

A lambda auditor still needs to be created.
